File: src/cpp/compile.cpp
#include <ctime>
#include <iomanip>
#include <filesystem>
#include <iostream>
using namespace std;


string to_4digit(int n) {
    string s = to_string(n);
    while (s.size() < 4) {
        s = "0" + s;
    }
    return s;
}


int main(){
    string s;
    int k, task_id;
    double mx_time, time_taken;
    clock_t start, end;
    #ifdef _WIN32
        system("g++ -o application main.cpp");
    #else
        system("g++ -o application.o main.cpp");
    #endif
    int i = 1;
    string input = "tests/" + to_4digit(i) + ".in";
    string output = "tests/" + to_4digit(i) + ".out";
    while(filesystem::exists(input)){
        #ifdef _WIN32
            s = ".\\application.exe < " + input + " > " + output;
        #else 
            s = "./application.o < " + input + " > " + output;
        #endif
        const char *arr = s.c_str();

        start = clock();
        int k = system(arr);
        end = clock();

        double time_taken = double(end - start)/double(CLOCKS_PER_SEC);

        if(k == 0){
            cout << "Test " << to_4digit(i) << " OK Time " << fixed << time_taken << setprecision(5) << " sec\n";

            if(mx_time < time_taken){
                task_id = i;
                mx_time = time_taken;
            }

        } else {
            cout << "Test " << to_4digit(i) << " ERROR Code " << k << endl;
        }
        ++i;
        input = "tests/" + to_4digit(i) + ".in";
        output = "tests/" + to_4digit(i) + ".out";
    }
    cout << "\nThe slowest task is " << to_4digit(task_id) << " Time: " << fixed << mx_time << setprecision(5) << " sec\n";
}

File: src/python/compile.py
import time
import subprocess
import os


s = ""
k, task_id = 0, 0
mx_time, time_taken = 0, 0
i = 1
input_file = f"tests/{i:04}.in"
output_file = f"tests/{i:04}.out"
while os.path.exists(input_file):
    if os.name == 'nt':
        s = f"py main.py < {input_file} > {output_file}"
    else:
        s = f"python3 main.py < {input_file} > {output_file}"

    start = time.time()
    k = os.system(s)
    end = time.time()

    time_taken = round(end - start, 5)
    if k == 0:
        print(f"Test {i:04} OK Time: {time_taken} sec")
        
        if mx_time < time_taken:
            task_id = i
            mx_time = time_taken
    else:
        print(f"Test {i:04} ERROR Code {k}")

    i += 1
    input_file = f"tests/{i:04}.in"
    output_file = f"tests/{i:04}.out"

print(f"\nThe slowest task is {task_id:04} Time: {mx_time} sec")
